addons/mail/models/mail_thread.py

```diff
@@ -1110,21 +1119,27 @@ def message_route(self, message, message_dict, model=None, thread_id=None, custo
         msg_references = [ref for ref in tools.mail_header_msgid_re.findall(thread_references) if 'reply_to' not in ref]
         mail_messages = MailMessage.sudo().search([('message_id', 'in', msg_references)], limit=1)
         is_a_reply = bool(mail_messages)
+        alias_domain = [('alias_name', 'in', rcpt_tos_localparts)]
 
         # 1.1 Handle forward to an alias with a different model: do not consider it as a reply
         if reply_model and reply_thread_id:
-            other_alias = Alias.search([
+            other_aliases = Alias.search([
                 '&',
                 ('alias_name', '!=', False),
-                ('alias_name', '=', email_to_localpart)
+                ('alias_name', 'in', email_to_localparts),
             ])
-            if other_alias and other_alias.alias_model_id.model != reply_model:
-                is_a_reply = False
+            for other_alias in other_aliases:
+                if other_alias.alias_model_id.model == reply_model:
+                    is_a_reply = bool(mail_messages)
+                    alias_domain.append(("alias_model_id.model", "=", reply_model))
+                    break
+                if other_alias.alias_model_id.model != reply_model:
+                    is_a_reply = False
 
         if is_a_reply:
             model, thread_id = mail_messages.model, mail_messages.res_id
             if not reply_private:  # TDE note: not sure why private mode as no alias search, copying existing behavior
-                dest_aliases = Alias.search([('alias_name', 'in', rcpt_tos_localparts)], limit=1)
+                dest_aliases = Alias.search(alias_domain, limit=1)
```

Comment:
Seems simpler to me to write

('alias_name', 'in', rcpt_tos_localparts),
('alias_model_id.model', '=', model)

As we want only aliases linked to the reply model, isn't it ? It should be the same as your code, but it is more explicit I think.

Reply:
OK I changed it.